monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/AppHelper.java

```diff
@@ -32,6 +37,11 @@ public IApp copyAppInfo(IApp app, AppForm appForm) {
         if (appForm.getRecipientIds() != null) {
             app.setRecipients(getRecipientsById(appForm.getRecipientIds()));
         }
+        if (!appForm.getUsername().isEmpty() && !appForm.getPassword().isEmpty()) {
+            app.setUsername(appForm.getUsername());
+            app.setPassword(passwordEncryptor.encrypt(appForm.getPassword()));
+           
+        }
```

Comment:
I wouldn't do that in the helper class. The manager should take care of it since it's business logic.